### PR TITLE
Enhance the .NET example

### PR DESCRIPTION
**Pull request by joaoasrosa**
This PR enhance the .NET example in the `sam init` examples:
* Adds the solution file
* Adds Cake, the DSL to build .NET solution
* Updates the README

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -611,6 +617,29 @@ file (or uber jar) containing all of the function dependencies.
...
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: HelloWorld::HelloWorld.Function::Handler
+      CodeUri: ./artifacts/HelloWorld.zip
+      Runtime: dotnetcore2.0
+```
+
+You should then build your ZIP file using your normal build process.
+
+You can find a .NET Core example in the `sam init` folder.
```
Comment (jfuss) [inline README.rst]: Can we change the wording here to be 'You can generate a .NET Core example by `sam init --runtime dotnetcore`'? While it is an example, its specific to cookiecutter and could be hard to parse. Generating the sample seems like a better experience to see and play with.

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-dotnet/{{cookiecutter.project_name}}/Solution.sln [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (jfuss) [inline samcli/local/init/templates/cookiecutter-aws-sam-hello-dotnet/{{cookiecutter.project_name}}/Solution.sln]: Should this have been committed? Looks like a ide generated file.

Comment (joaoasrosa) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-dotnet/{{cookiecutter.project_name}}/Solution.sln]: @jfuss in dotnet world the Solution file "glues" the projects. It is an artifact used to make it easy to restore NuGet packages or build the projects in one go. It is an abstraction created by Microsoft.
Started with Visual Studio, but if you use JetBrains Rider, it recognises and uses it.

When you are using Visual Studio Code you do not need it, since VS Code opens a folder.

**Comment by jfuss** [comment]
@joaoasrosa This is really awesome! Thanks for doing it and then redoing it :). I am going to hold off and merge for right now. I want to pull down the commit(s) and run locally, as a sanity test (we don't have deep tests for any of this yet). Unfortunately, I am not in a spot to pull this down right now. I may have some time tomorrow when I get back from PyCon. If not, I will handle Wednesday for sure.

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-dotnet/{{cookiecutter.project_name}}/Solution.sln (hunk 2 above)
Comment (joaoasrosa) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-dotnet/{{cookiecutter.project_name}}/Solution.sln]: No problem. Sharing is caring! ;)

**Comment by joaoasrosa** [comment]
@jfuss please let me know if I miss anything. Cheers

File: README.rst [hunk 3]
```diff
@@ -611,6 +617,29 @@ file (or uber jar) containing all of the function dependencies.
...
+To use SAM Local with compiled languages, such as .NET Core that require a packaged artifact (e.g. a ZIP), you can specify the location of the artifact with the `AWS::Serverless::Function` `CodeUri` property in your SAM template.
+
+For example:
+
+```
+AWSTemplateFormatVersion: 2010-09-09
+Transform: AWS::Serverless-2016-10-31
+
+Resources:
+  ExampleJavaFunction:
```
Comment (heitorlessa) [inline README.rst]: When possible, mind changing to ExampleDotNetFunction?

**Comment by heitorlessa** [comment]
That is awesome @joaoasrosa, thank you so much for contributing to that! I knew there was a better way (Cake) but wasn't sure what .Net folks do these days -- Much appreciated!

**Comment by joaoasrosa** [comment]
@heitorlessa thanks for pointing out. :)
Cake makes thing easier. I'm baking a more comprehensive example with containers. ;)

**Comment by joaoasrosa** [comment]
@jfuss did you confirm the solution in the PR?

**Comment by jfuss** [comment]
@joaoasrosa I did. Thanks for the reminder ping.
